### PR TITLE
chore: Improve ts third-party tests

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

```
 ✔ 3303 valid programs parsed without error
 ✔ 409 invalid programs produced a parsing error
 ✔ 1955 invalid programs did not produce a parsing error (and allowed by the allowlist file)
 ✔ 10 valid programs produced a parsing error (and allowed by the allowlist file)
```
We don't do type checking, so it's normal to have a lot of `invalid programs did not produce a parsing error`.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14939"></a>

### CODE REVIEW

File: scripts/parser-tests/typescript/index.js [hunk 1]
```diff
@@ -2,15 +2,12 @@ import path from "path";
 import fs from "fs";
 import { fileURLToPath } from "url";
 import TestRunner from "../utils/parser-test-runner.js";
-import ErrorCodes from "./error-codes.js";
```
Comment (nicolo-ribaudo) [inline scripts/parser-tests/typescript/index.js:5]: Why this change?

Comment (liuxingbaoyu) [reply scripts/parser-tests/typescript/index.js:5]: This used to be used for diagnostic information generated by `typescript`, but was changed to detect the `.errors.txt` file in a PR half a year ago.
So now we don't need it anymore, but I didn't delete this file in case there is any need in the future.

**Comment by liuxingbaoyu** [comment]
I now think I should keep this filtering, it just needs a little modification. 😕
